packages/neuron-wallet/src/services/node.ts

```diff
@@ -138,13 +143,66 @@ class NodeService {
       return false
     } catch (err) {
       logger.info('CKB:\texternal RPC on default uri not detected, starting bundled CKB node.')
-      return startCkbNode().then(()=> true).catch(err => {
-        logger.info('CKB:\tfail to start bundled CKB with error:')
-        logger.error(err)
-        return false
-      })
+      const isReadyToStart = await this.detectDependency()
+      return isReadyToStart ? this.startNode() : this.showGuideDialog()
     }
   }
+
+  private showGuideDialog = () => {
+    const I18N_PATH = `messageBox.ckb-dependency`
+    return dialog.showMessageBox({
+      type: 'info',
+      buttons: ['skip', 'install-and-exit'].map(label => i18n.t(`${I18N_PATH}.buttons.${label}`)),
+      defaultId: 1,
+      title: i18n.t(`${I18N_PATH}.title`),
+      message: i18n.t(`${I18N_PATH}.message`),
+      detail: i18n.t(`${I18N_PATH}.detail`),
+      cancelId: 0,
+      noLink: true,
+    }).then(({ response }) => {
+      switch (response) {
+        case 1: {
+          // open browser and shutdown
+          const VC_REDIST_URL = `https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads`
+          shell.openExternal(VC_REDIST_URL)
+          env.app.quit()
+          return false
+        }
+        case 0:
+        default: {
+          // dismiss dialog and continue
+          return this.startNode()
+        }
+      }
+    })
+  }
+
+  private detectDependency = async () => {
+    if (process.platform !== 'win32') {
+      return true
+    }
+    const execPromise = promisify(exec)
+    const arches = process.arch === 'x64' ? ['x86', 'x64'] : ['x64']
+    const queries = arches.map(arch =>
+      `REG QUERY ` +
+      [`HKEY_LOCAL_MACHINE`, `SOFTWARE`, `Microsoft`, `VisualStudio`, `14.0`, `VC`, `Runtimes`, arch].join(path.sep))
```

Comment:
Not sure if I got it right.
What if the link `https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads` provides the download for version `15.0`? Will the dialog keep prompting because it checks against version `14`?

Reply:
It does have a potential problem and the hardcoded query is not ideal.

For now, it's just a workaround cuz the version specified in the documentation is 2019, which shares the same files with 2015/2017 and they are all located in the `VisualStudio/14/...`.

We should update the query along with the version specified in the documentation and perhaps we should make it automatic in the CI.

Speaking of the question

> What happens if the link https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads provides the download for version 15.0? Will the dialog keep prompting because it checks against version 14?

For now, we should still check the `14`, and users should pick the `vc_redist 2015-2019` as the documentation says.

Reply:
We may also need to consider the cases for the older versions of windows, such as Win7, which is not supported officially anymore. I am wondering if the latest `vc_redist` is supported on such legacy windows versions. If they are not supported, meaning users can only install older versions of `vc_redist`, which would prompt the dialog every time the Neuron starts?

Do we know the lowest version of `vc_redist` that CKB node requires? If so, maybe we can loop and check the existence of registres from the lowest to highest version. This way may provide a bit more compatibility over combinations of different windows and `vc_redist` versions.

Reply:
The consideration does make sense.
The vc_redist 2015-2019 works well on Win7 thanks to the awesome backward-compatibility of Windows.
And the officially specified by CKB is the 2019 one

> (minimum required) Or you can just select two individual components: "MSVC v142 - VS 2019 C++ x64/x86 build tools (vXX.XX)" and "Windows 10 SDK (10.0.X.0)".

Reply:
Sounds good.
Let's keep our eyes on the newer versions of `vs_redist` in the future, because the newer version could bring in higher number than `14`, which would unnecessarily trigger the dialog. At that stage, we need to refactor the code to make it more adaptable if we decided to leave it for now.